Sources/UtilityBeltNetworking/HTTPClient.swift

```diff
@@ -95,32 +94,37 @@ public class HTTPClient {
     /// - Parameter url: The URL for the request. Accepts a URL or a String.
     /// - Parameter method: The HTTP method for the request.
     /// - Parameter parameters: The dictionary of parameters to send in the query string or HTTP body.
+    /// - Parameter headers: The HTTP headers to be with the request.
     /// - Parameter encoding: The parameter encoding method. If nil, uses default for HTTP method.
-    /// - Parameter completion: The completion block to call when the request is completed, regardless of error.
+    /// - Parameter jsonDecoder: The `JSONDecoder` to use when decoding the response data.
+    /// - Parameter completion: The completion block to call when the request is completed.
     @discardableResult
     public func request<T>(_ url: URLConvertible,
                            method: HTTPMethod,
                            parameters: [String: Any]? = nil,
+                           headers: HTTPHeaderDictionaryConvertible? = nil,
                            encoding: ParameterEncoding? = nil,
+                           decoder: JSONDecoder = JSONDecoder(),
                            completion: DecodableTaskCompletion<T>? = nil) -> URLSessionTask? where T: Decodable {
-        guard let url = try? url.asURL() else {
-            let result: Result<T, Error> = .failure(UBNetworkError.unexpectedError)
-            let response = DataResponse(request: nil, response: nil, data: nil, result: result)
-            completion?(response)
-            return nil
-        }
-
-        return self.request(url, method: method, parameters: parameters, encoding: encoding) { dataResponse in
+        return self.request(
+            url,
+            method: method,
+            parameters: parameters,
+            headers: headers,
+            encoding: encoding
+        ) { dataResponse in
             // TODO: Check the response.mimeType and ensure it is application/json, which is required for decoding
 
             // Create a result object for improved handling of the response
             let result: Result<T, Error> = {
                 switch dataResponse.result {
                 case let .success(data):
-                    if let decodedObject = try? JSONDecoder().decode(T.self, from: data) {
+                    do {
+                        let decodedObject = try decoder.decode(T.self, from: data)
                         return .success(decodedObject)
-                    } else {
-                        return .failure(UBNetworkError.unableToDecode(String(describing: T.self)))
+                    } catch {
+//                            return .failure(UBNetworkError.unableToDecode(String(describing: T.self)))
```

Comment:
Is this comment important to keep around?

Reply:
Nope, should have been deleted. Thanks!